lib/ansible/modules/identity/cyberark/cyberark_user.py

```diff
@@ -405,54 +412,58 @@ def user_add_to_group(module):
-        "state": {"type": "str",
-                  "choices": ["present", "absent"],
-                  "default": "present"},
-        "cyberark_session": {"required": True, "type": "dict"},
-        "initial_password": {"type": "str", "no_log": True},
-        "new_password": {"type": "str", "no_log": True},
-        "email": {"type": "str"},
-        "first_name": {"type": "str"},
-        "last_name": {"type": "str"},
-        "change_password_on_the_next_logon": {"type": "bool"},
-        "expiry_date": {"type": "str"},
-        "user_type_name": {"type": "str"},
-        "disabled": {"type": "bool"},
-        "location": {"type": "str"},
-        "group_name": {"type": "str"},
-    }
-
-    module = AnsibleModule(argument_spec=fields)
-
-    state = module.params["state"]
-
-    changed = False
-    result = {}
+    module = AnsibleModule(
+        argument_spec=dict(
+            username=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['absent', 'present']),
+            cyberark_session=dict(type='dict', required=True),
+            initial_password=dict(type='str', no_log=True),
+            new_password=dict(type='str', no_log=True),
+            email=dict(type='str'),
+            first_name=dict(type='str'),
+            last_name=dict(type='str'),
+            change_password_on_the_next_logon=dict(type='bool'),
+            expiry_date=dict(type='str'),
+            user_type_name=dict(type='str'),
+            disabled=dict(type='bool'),
+            location=dict(type='str'),
+            group_name=dict(type='str'),
+        ),
+    )
+
+    state = module.params['state']
+    new_password = module.params['new_password']
+    group_name = module.params['group_name']
+
+    result = dict(
+        changed=False,
+    )
 
     if (state == "present"):
-        (changed, result, status_code) = user_details(module)
-        if (status_code == 200):  # user already exists
-            if ("new_password" in module.params):
-                # if new_password specified, proceed to update user credential
-                (changed, result, status_code) = user_add_or_update(module, "PUT")
-            if ("group_name" in module.params and module.params["group_name"] is not None):
-                # if user exists, add to group if needed
-                (changed, ignored_result, ignored_status_code) = user_add_to_group(module)
-        elif (status_code == 404):
-            # user does not exist, proceed to create it
-            (changed, result, status_code) = user_add_or_update(module, "POST")
-            if (status_code == 201 and "group_name" in module.params and module.params["group_name"] is not None):
-                # if user was created, add to group if needed
-                (changed, ignored_result, ignored_status_code) = user_add_to_group(module)
+        (result['changed'], result['cyberark_user'], result['status_code']) = user_details(module)
```

Comment:
What `validate-modules` rule is all this trying to fix? This looks more like personal code style changes (and I personally find the original style easier to read, since there's only one exit point that needs to update the result)

Reply:
Indeed, this one should not have been part of this PR.